src/r3/atomic-swap/corda/evm-bridge-contracts/src/main/kotlin/com/r3/corda/DefaultEventEncoder.kt

```diff
@@ -0,0 +1,117 @@
+package com.r3.corda.evmbridge
+
+import com.r3.corda.cno.evmbridge.dto.TransactionReceipt
+import com.r3.corda.cno.evmbridge.dto.TransactionReceiptLog
+import net.corda.core.serialization.CordaSerializable
+import org.web3j.abi.DefaultFunctionEncoder
+import org.web3j.crypto.Hash
+import org.web3j.abi.TypeEncoder
+import org.web3j.abi.datatypes.Type
+import org.web3j.abi.datatypes.*
+import org.web3j.abi.datatypes.generated.*
+import org.web3j.utils.Numeric
+import java.io.Serializable
+import java.math.BigInteger
+
+data class Indexed<T>(val indexedValue: T)
+
+class DefaultEventEncoder {
+    fun encodeEvent(contractAddress: String, eventSignature: String, vararg params: Any): EncodedEvent {
+        val paramTypesString = eventSignature.substringAfter('(').substringBefore(')')
+        val paramTypes = paramTypesString.split(',').map { it.trim() }
+
+        fun toWeb3jType(value: Any, type: String): Pair<Type<out Serializable>, Boolean> {
+            val isIndexed = value is Indexed<*>
+            val unwrappedValue = if (isIndexed) (value as Indexed<*>).indexedValue else value
+
+            return Pair(when (type) {
+                "string" -> Utf8String(unwrappedValue as String)
+                "uint256" -> Uint256(unwrappedValue as BigInteger)
+                "uint8" -> Uint8(unwrappedValue as BigInteger)
+                "int256" -> Int256(unwrappedValue as BigInteger)
+                "address" -> Address(unwrappedValue as String)
+                "bool" -> Bool(unwrappedValue as Boolean)
+                "bytes" -> DynamicBytes(unwrappedValue as ByteArray)
+                else -> throw IllegalArgumentException("Unsupported type: $type")
+            }, isIndexed)
+        }
+
+        val web3jParamsWithIndexedInfo = params.zip(paramTypes).map { (value, type) -> toWeb3jType(value, type) }
+
+        val indexedParams = web3jParamsWithIndexedInfo.filter { it.second }.map { it.first }
+        val nonIndexedParams = web3jParamsWithIndexedInfo.filterNot { it.second }.map { it.first }
+
+        val topic0 = Hash.sha3String(eventSignature.replace("\\s+".toRegex(), ""))
+        val topics = listOf(topic0) + indexedParams.map { Numeric.prependHexPrefix(TypeEncoder.encode(it)) }
+        val data = Numeric.prependHexPrefix(DefaultFunctionEncoder().encodeParameters(nonIndexedParams))
+
+        return EncodedEvent(contractAddress, topics, data)
+    }
+}
+
+@CordaSerializable
+data class EncodedEvent(
+    val address: String,
+    val topics: List<String>,
+    val data: String
+) {
+    companion object {
+        val defaultLog = TransactionReceiptLog()
+    }
+
+    data class Log(val found: Boolean, val log: TransactionReceiptLog)
```

Comment:
This could use some KDocs as its a bit obscure what "found" means in the context of a Log class